src/fermilib/circuits/low_depth_trotter_simulation_test.py

```diff
@@ -0,0 +1,433 @@
+            self.register, hamiltonian, trotter_steps=2, first_order=False)
+
+        self.engine.flush()
+
+        # get_sparse_operator reverses the indices, so reverse the sites
+        # the Hamiltonian acts on so as to compare them.
+        reversed_operator = (0.37 * FermionOperator('3^ 2^ 3 2') +
+                             FermionOperator('3^ 1') + FermionOperator('1^ 3'))
+        evol_matrix = expm(-1j * get_sparse_operator(
+            reversed_operator, n_qubits=self.size)).todense()
+        expected = evol_matrix * numpy.matrix([2 ** (-self.size / 2.)] *
+                                              2 ** self.size).T
+
+        self.assertTrue(numpy.allclose(ordered_wavefunction(self.engine),
+                                       expected.T,
+                                       atol=1e-2))
+
+    def test_simulate_dual_basis_hamiltonian(self):
+        hamiltonian = dual_basis_hamiltonian(1, self.size)
+        self.engine.flush()
+
+        # Choose random state.
+        initial_state = numpy.zeros(2 ** self.size, dtype=complex)
+        for i in range(len(initial_state)):
+            initial_state[i] = (random.random() *
+                                numpy.exp(1j * 2 * numpy.pi * random.random()))
+        initial_state /= numpy.linalg.norm(initial_state)
+
+        # Put randomly chosen state in the registers.
+        self.engine.backend.set_wavefunction(initial_state, self.register)
+
+        low_depth_trotter_simulation.simulate_dual_basis_evolution(
+            self.register, hamiltonian, trotter_steps=7, first_order=False)
+
+        self.engine.flush()
+
+        # get_sparse_operator reverses the indices, but the jellium
+        # Hamiltonian is symmetric.
+        evol_matrix = expm(-1j * get_sparse_operator(
+            hamiltonian, n_qubits=self.size)).todense()
+        initial_state = numpy.matrix(initial_state).T
+        expected = evol_matrix * initial_state
+
+        self.assertTrue(numpy.allclose(ordered_wavefunction(self.engine),
+                                       expected.T,
+                                       atol=1e-2))
+
+    def test_simulate_dual_basis_evolution_bad_input_ordering(self):
+        with self.assertRaises(ValueError):
+            low_depth_trotter_simulation.simulate_dual_basis_evolution(
+                self.register, FermionOperator(), input_ordering=[1, 2])
+
+    def test_simulate_dual_basis_evolution_n_trotter_steps_not_integer(self):
+        with self.assertRaises(ValueError):
+            low_depth_trotter_simulation.simulate_dual_basis_evolution(
+                self.register, FermionOperator(), trotter_steps=1.5)
+
+    def test_simulate_dual_basis_evolution_bad_n_trotter_steps(self):
+        with self.assertRaises(ValueError):
+            low_depth_trotter_simulation.simulate_dual_basis_evolution(
+                self.register, FermionOperator(), trotter_steps=0)
+
+
+class FourQubitFirstOrderEquivalenceWithSecondOrderTest(unittest.TestCase):
```

Comment:
The space isn't actually required by PEP8 - I've changed it for better consistency with the rest of the package (though tbh I personally don't like it)